bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/channel/ChannelTypeResource.java

```diff
@@ -137,8 +129,8 @@ private ChannelTypeDTO convertToChannelTypeDTO(ChannelType channelType, Locale l
                 parameterGroups, channelType.getState(), channelType.getTags(), channelType.isAdvanced());
     }
 
-    private Set<ChannelTypeDTO> convertToChannelTypeDTOs(List<ChannelType> channelTypes, Locale locale) {
-        Set<ChannelTypeDTO> channelTypeDTOs = new HashSet<>();
+    private List<ChannelTypeDTO> convertToChannelTypeDTOs(List<ChannelType> channelTypes, Locale locale) {
+        List<ChannelTypeDTO> channelTypeDTOs = new ArrayList<>();
```

Comment:
Looks like `ChannelTypeDTO` has no own hasCode or equals method. So the Set would have had a hard time finding duplicates in the first place.
How likely is a setup where `channelTypeRegistry.getChannelTypes(locale)` returns duplicate channelTypes?